roles/common/tasks/iptables.yml

```diff
@@ -0,0 +1,76 @@
+---
+- name: Disable firewalld
+  service:
+    name: firewalld
+    state: stopped
+    enabled: false
+  failed_when: false
+
+- name: Install iptables
+  package: name=iptables
+  register: result
+  until: result|succeeded
+  retries: 5
+
+- name: Install iptables-services
+  package: name=iptables-services
+  register: result
+  until: result|succeeded
+  retries: 5
+
+- name: Enable iptables
+  service:
+    name: iptables
+    state: started
+    enabled: true
+
+- name: ensure iptables allows established and related traffic
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*INPUT.*ESTABLISHED,RELATED.*ACCEPT"
+    insertafter: "^:OUTPUT "
+    line: "-A INPUT -m state --state ESTABLISHED,RELATED -j ACCEPT"
+    backup: yes
+  notify: restart iptables
+
+- name: Permit SSH
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*INPUT.*{{ item.protocol }}.*{{ item.port }}.*ACCEPT"
+    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
+    line: "-A INPUT -m state --state NEW -m {{ item.protocol }} -p {{ item.protocol }} --dport {{ item.port }} -j ACCEPT"
+    backup: yes
+  with_items:
+    - protocol: tcp
+      port: 22
+  notify: restart iptables
+
+- name: Permit SSH For Sources
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*INPUT.*{{ item.protocol }}.*-s {{ item.source }}.*{{ item.port }}.*ACCEPT"
+    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
+    line: "-A INPUT -m state --state NEW -m {{ item.protocol }} -p {{ item.protocol }} -s {{ item.source }} --dport {{ item.port }} -j ACCEPT"
+    backup: yes
+  with_items:
+    - source: "{{ common.ssh.allow_from }}"
```

Comment:
It failed when running on rhel vms, and here is my solution.

- name: Permit SSH For Sources
  lineinfile:
    dest: /etc/sysconfig/iptables
    state: present
    regexp: "^.*INPUT.*{{ item[1] }}.*-s {{ item[0]}}.*{{ item[2] }}.*ACCEPT"
    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
    line: "-A INPUT -m state --state NEW -m {{ item[1] }} -p {{ item[1] }} -s {{ item[0] }} --dport {{ item[2] }} -j ACCEPT"
    backup: yes
  with_nested:
    - "{{ common.ssh.allow_from }}"
    - tcp
    - 22
  notify: restart iptables